CMPUT404W22AMNRY/CMPUT404-project-socialdistribution
# Add rest endpoint for authors

### Overview
* Install django-rest-framework
* Implement `/api/v1/authors/` GET endpoint
* Use custom paginator to match the interface given in the requirements

### Notes
There are helper classes under `api/util` to match the JSON format specified under the project requirements (i.e. have a `type` field and render results under `items`

Sample response
```json
{
  "type": "authors",
  "items": [
    {
      "id": 1,
      "url": "http://localhost:8000/api/v1/authors/1/",
      "type": "author",
      "displayName": "asdf asdf",
      "github": "asdf",
      "profileImage": "asdf"
    }
  ]
}
```

### Future work
Do the same for other resources (posts, comments, etc.)

### How to test
1. Create a new user from http://localhost:8000/accounts/signup
2. Create a new admin user `./manage.py createsuperuser admin`
3. Activate the user from step 1 by
    1. Navigating to http://localhost:8000/admin
    2. Click on the user created from step 1 and set the its `is_active` status to active
4. Use curl to verify the response `curl localhost:8000/api/v1/authors/ -u username:password`. The username can be the user from step 1 or the admin from step 2

Part of #32
